From 72f2c447f6849bf2f5fbc3ce1eb61adb18126de6 Mon Sep 17 00:00:00 2001
From: Lee Treveil <leetreveil@gmail.com>
Date: Mon, 2 May 2016 19:19:27 +0100
Subject: [PATCH] Don't use instanceof to test stream is a stream

---
 lib/browser.js | 4 ++--
 package.json   | 1 +
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/browser.js b/lib/browser.js
index 8ffc207..014815c 100644
--- a/lib/browser.js
+++ b/lib/browser.js
@@ -3,7 +3,7 @@
 var readStream = require('filereader-stream')
 var through = require('through')
 var musicmetadata = require('./index')
-var Stream = require('stream').Stream
+var isStream = require('is-stream')
 
 module.exports = function (stream, opts, callback) {
   return musicmetadata(wrapFileWithStream(stream), opts, callback)
@@ -24,7 +24,7 @@ function wrapFileWithStream (file) {
     })
   }
 
-  if (file instanceof Stream) {
+  if (isStream(file)) {
     return file.pipe(stream)
   }
   if (file instanceof window.FileList) {
diff --git a/package.json b/package.json
index c659e82..f079ba4 100644
--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
   "dependencies": {
     "deep-equal": "0.2.1",
     "filereader-stream": "^0.2.0",
+    "is-stream": "^1.1.0",
     "strtok2": "~1.0.0",
     "sum-component": "^0.1.1",
     "through": "~2.3.4"
